monai/data/grid_dataset.py

```diff
@@ -20,64 +20,143 @@
 from monai.transforms import apply_transform
 from monai.utils import NumpyPadMode, ensure_tuple
 
-__all__ = ["PatchDataset", "GridPatchDataset"]
+__all__ = ["PatchDataset", "GridPatchDataset", "PatchIter"]
 
 
-class GridPatchDataset(IterableDataset):
+class PatchIter:
```

Comment:
It's worth noting this whole class could be substituted with use of `partial` with `iter_patch` like:

```python
patch_iter=partial(iter_patch, patch_size=..., start_pos=..., copy_back=False, mode=..., **pad_opts)
```

Reply:
yes I was using the partial function, it seems it's not compatible with the multiprocessing loader, the error message said it's not a picklable object

Reply:
It's going to depend on the arguments passed into partial, I can get it to pickle with simple values and a simple function but something more complex or locally declared would fail.